test/crawlers/browser_crawler.test.js

```diff
@@ -362,7 +362,10 @@ describe('BrowserCrawler', () => {
     });
 
     test('should persist cookies per session', async () => {
+        const name = `list-${Math.random()}`;
         const requestList = new Apify.RequestList({
+            persistStateKey: name,
+            persistRequestsKey: name,
```

Comment:
Are you really sure this is in memory and not using a default name? Because the test was failing, and I was expect it to be failing because of using the same default name for storing the list. Already fixed one test the very same way. 

Ideally we should have all the tests idendendent on external services, but when I tried to run tests without internet connection, _a lot_ were failing, to get around all of them would take probably more than a week.

Reply:
The failure is here, maybe you have better idea on what this can be caused by:

```
FAIL test/crawlers/browser_crawler.test.js (17.643 s)
  ● BrowserCrawler › should persist cookies per session

    expect(received).toEqual(expected) // deep equality

    Expected: "TEST=12321312312"
    Received: ""

      395 |         await browserCrawler.run();
      396 |         expect(loadedCookies).toHaveLength(4);
    > 397 |         loadedCookies.forEach((cookie) => expect(cookie).toEqual('TEST=12321312312'));
          |                                                          ^
      398 |     });
      399 |
      400 |     test('should throw on "blocked" status codes', async () => {

      at forEach (test/crawlers/browser_crawler.test.js:397:58)
          at Array.forEach (<anonymous>)
      at Object.<anonymous> (test/crawlers/browser_crawler.test.js:397:23)
```

Reply:
I don't know to be honest, but your earlier fix did not work for the other test. It failed for me later anyway. I'm like 99.9% sure it has nothing to do with the request list persistence.

In the earlier test, I just added this and I'm waiting for it to fail again. Which it has not done since. Obviously.
```js
sessions.forEach((session) => {
                // TODO this test is flaky in CI and we need some more info to debug why.
                if (session.errorScore !== 1) {
                    console.log('SESSIONS:');
                    console.dir(sessions);
                }

                expect(session.errorScore).toEqual(1);
            });
```

Reply:
> Ideally we should have all the tests idendendent on external services, but when I tried to run tests without internet connection, a lot were failing, to get around all of them would take probably more than a week.

This is true, but I think most of them are now pointing to `example.com` which is very reliable, so it's most likely not a problem of the external service in all cases where we open `example.com`. Low hanging fruit to fix are all the cases that do not point to `example.com`